opts.go

```diff
@@ -47,3 +48,14 @@ func WithProcessRunner(cmd *exec.Cmd) Opt {
 		machine.cmd = cmd
 	}
 }
+
+// WithSnapshot will allow for the machine to start using a given snapshot.
+func WithSnapshot(memFilePath, snapshotPath string, opts ...LoadSnapshotOpt) StartOpt {
+	return func(m *Machine) {
+		m.Cfg.Snapshot.MemFilePath = memFilePath
+		m.Cfg.Snapshot.SnapshotPath = snapshotPath
+		m.Cfg.Snapshot.Opts = opts
```

Comment:
We technically could, I just feel it would clutter the code here when it can be done much cleaner in [firecracker.go](https://github.com/sondavidb/firecracker-go-sdk/blob/snapshot-tester/firecracker.go?#L290) and makes it more consistent with other functions in that file. Doing it here would require having the user re-provide the context they're already using for Start as well as require us to import context.

Reply:
I don't get the context part. Applying options means just mutating m.Cfg.Snapshot (or LoadSnapshotParams).

Reply:
Ah, I see, my bad. I agree with the idea, but this would require the WithSnapshot opt to take in a different option type, i.e. make a new option type in opts.go or machine.go and trash LoadSnapshotOpt in firecracker.go. I feel given all the other functions in firecracker.go it would be inconsistent to have one function that doesn't have or use the opt type. But if you think it makes more sense to handle these options within opts.go, I can certainly do that.

Reply:
Do you have a function to modify EnableDiffSnapshots? Shouldn't it take SnapshotConfig?

Reply:
We looked into it a bit more, and while this is doable via a new option type, the main caveat is that it removes the user's ability to set some client-side configurations when reaching firecracker.go (e.g. WithContext(), WithTimeout(), etc). However, not being able to change the variables (timeout, Context, and HTTPClient) in the scope of LoadSnapshot sounds fine to me? But I'm not really sure — what do you think? Is this alright?